GetandTakeAPI/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using GetandTake.Business.Services.Abstract;
+using GetandTake.Models.DTOs.ListDTO;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GetandTakeAPI.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class ProductsController : ControllerBase
+{
+    private readonly IProductService _productService;
+
+    public ProductsController(IProductService productService)
+    {
+        _productService = productService;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<List<ProductsDTO>>> GetAllProducts()
+    {
+        var products = await _productService.GetAllAsync();
+
+        return Ok(products);
+    }
+
+    [HttpGet("id/{id:int}")]
+    public async Task<ProductsDTO> GetProductByIdAsync(int id)
```

Comment:
Please, parametrise it with `Task<ActionResult<ProductsDTO>>`.

BTW, `ProductsDTO ` is not a really correct name since it is plural, but your entity is single. Make sure that name of DTO has singular form

Reply:
I need to rename `ProductsDTO` :)